File: app/src/main/java/com/github/komarnicki/thomas/ringtimer/model/Timer.java
package com.github.komarnicki.thomas.ringtimer.model;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;
import android.os.Parcel;
import android.os.Parcelable;

@Entity
public class Timer implements Parcelable{

    @PrimaryKey(autoGenerate = true)
    private int id;

    private int duration;

    private int breakTime;

    @Ignore
    public Timer(int duration, int breakTime) {
        this.duration = duration;
        this.breakTime = breakTime;
    }

    public Timer() {

    }

    protected Timer(Parcel in) {
        id = in.readInt();
        duration = in.readInt();
        breakTime = in.readInt();
    }

    public static final Creator<Timer> CREATOR = new Creator<Timer>() {
        @Override
        public Timer createFromParcel(Parcel in) {
            return new Timer(in);
        }

        @Override
        public Timer[] newArray(int size) {
            return new Timer[size];
        }
    };

    public int getDuration() {
        return duration;
    }

    public void setDuration(int duration) {
        this.duration = duration;
    }

    public int getBreakTime() {
        return breakTime;
    }

    public void setBreakTime(int breakTime) {
        this.breakTime = breakTime;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(id);
        dest.writeInt(duration);
        dest.writeInt(breakTime);
    }
}

File: app/src/main/java/com/github/komarnicki/thomas/ringtimer/model/Timer.kt
package com.github.komarnicki.thomas.ringtimer.model

import android.arch.persistence.room.Entity
import android.arch.persistence.room.Ignore
import android.arch.persistence.room.PrimaryKey
import android.net.Uri
import android.os.Parcel
import android.os.Parcelable
import com.github.komarnicki.thomas.ringtimer.R
import com.github.komarnicki.thomas.ringtimer.createParcel

@Entity
class Timer : Parcelable {

    @PrimaryKey(autoGenerate = true)
    var id: Int = 0

    var duration: Int = 0

    var breakTime: Int = 0

    var warning: Int = -1

    @Ignore
    private var soundClip: String = "android.resource://com.github.komarnicki.thomas.ringtimer/raw/bell"

    var soundClipUri: Uri
    get() {return Uri.parse(soundClip)}
    set(value) {soundClip = value.toString()}

    @Ignore
    constructor(duration: Int, breakTime: Int) {
        this.duration = duration
        this.breakTime = breakTime
    }

    constructor()

    protected constructor(`in`: Parcel) {
        id = `in`.readInt()
        duration = `in`.readInt()
        breakTime = `in`.readInt()
        warning = `in`.readInt()
        soundClip = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(id)
        dest.writeInt(duration)
        dest.writeInt(breakTime)
        dest.writeInt(warning)
        dest.writeString(soundClip)
    }



    companion object {
        @JvmField @Suppress("unused")
        val CREATOR = createParcel {Timer(it)}
    }

    fun totalSecondsDuration(): Int{
        return duration + breakTime
    }

}

//@TypeConverter fun uriToString(uri: Uri): String{
//    return uri.toString()
//}
//@TypeConverter fun stringToUri(s: String): Uri{
//    return Uri.parse(s)
//}